connectOpenAIAPI.php

```diff
@@ -1,50 +1,87 @@
 <?php
+    header("Access-Control-Allow-Origin: *");
+    header("Access-Control-Allow-Methods: GET, POST, OPTIONS");
+    header("Access-Control-Allow-Headers: Content-Type");
+
     include("include/init.php");
     require_once("connectYTAPI.php");
     include("env_constants.php");
 
-    // call function to retrieve video data from YouTube API
-    $youtubeData = getYoutubeData();
+    acceptCorsRequestsFromTheChromeExtension();
 
-    // array that will hold messages to ask ChatGPT
-    $input = [];
+    // save videoId, ThemesToReview, and RangeSlider from GET request as variables
+    $videoId = $_GET['videoId'];
+    $ThemesToReviewJSON = $_GET['reviewThemes'];
+    $ThemesToReview = json_decode($ThemesToReviewJSON, true);
+    $RangeSliderJSON = $_GET['rangeSlider'];
+    $RangeSlider = json_decode($RangeSliderJSON, true);
 
-    foreach($youtubeData as $video){
-        $title = $video[0];
-        $description = $video[1];
-        $categoryId = $video[2];
-        $tags = $video[3];
-        $stringTags = implode(',', $tags);
-        $rating = $video[4];
 
-        array_push($input, "The video has a title of '".$title."' a description of '".$description."' a category id of ".$categoryId." a rating of ".$rating." and the following tags: ".$stringTags);
+    // retrieves video data from YouTube API and returns an array with important content
+    $video = getYoutubeData($videoId);
 
-    }
+    $title = $video['title'];
+    $description = $video['description'];
+    $categoryId = $video['categoryId'];
+    $tags = $video['tags'];
+    $stringTags = implode(',', $tags);
```

Comment:
If you think this is always how you'll want the tags returned, you could do this inside the `getYoutubeData` function. You probably *also* want the existing `tags` format (where it's an array), but if you did this implode inside the function, it'd clean up the code whenever you need to use this data.